include/fs/fs.h

```diff
@@ -170,20 +182,32 @@ struct fs_file_system_t {
 #define FS_SEEK_END	2	/* Seek from end of file.  */
 #endif
 
-
 /**
- * @brief File open
+ * @brief File open extended
  *
- * Opens an existing file or create a new one and associates
- * a stream with it.
+ * Opens or crates, if does not exist file, depending on mode provided
+ * and associates a stream with it.
  *
  * @param zfp Pointer to file object
  * @param file_name The name of file to open
+ * @param flags The mode flags, if not provided default to FO_CREATE | FO_RDWR
+ *
+ * @p flags can be empty, or combination of one or more of following flags:
+ *   FO_READ		open for read
+ *   FO_WRITE		open for write
+ *   FO_RDWR		open for read/write (<tt>FO_READ | FO_WRITE</tt>)
+ *   FO_CREATE		crete file if it does not exist
+ *   FO_APPEND		open file at the end
  *
  * @retval 0 Success
- * @retval -ERRNO errno code if error
+ * @retval -EINVAL when bad combination of flags is given
+ *	   -EINVAL when bad file name is given
+ *	   -NOENT  when file path is not possible (bad mount point)
+ *	   -ERRNO  other errno code, deending of file system backend
  */
-int fs_open(struct fs_file_t *zfp, const char *file_name);
+int fs_open(struct fs_file_t *zfp, const char *file_name, unsigned int flags);
+#define fs_open(f, ...) _fs_open(f, __VA_ARGS__, FO_CREATE | FO_RDWR)
```

Comment:
Ah, so there's actually a magic to make older 2-arg fs_open() to work with the new 3-arg fs_open()? A bit too much magic to me, I hope somebody vetted that.

Reply:
Yes, I have not found a way to resolve this with C syntax only. Maybe someone does know about some extension into gcc and clang that would fix this. There is less magic here than in `OPEN_CHOOSE`, which is more advance usage of cpp than the two cpp lines used here.

Reply:
> Yes, I have not found a way to resolve this with C syntax only. Maybe ...

I wouldn't worry too much about it, if it works, it's ok. We should treat that as as a deprecated transitional feature, to be removed eventually.

Reply:
It has been decided to drop backward compatibility and note need for changes, in existing code, in release notes.